RELEASING.md

```diff
@@ -0,0 +1,241 @@
+upper bound on `containers` package, because that's just dumb. On the other hand all of
+the dependeices that come from CHaPs should follow strict bounds, because those packages
+are a lot more volatile and creating revisions for packages on CHaPs is not as smooth as
+it is on Hackage.
+
+### `CHANGELOG.md`
+
+Every package, that [is subject to the release
+process](packages-excluded-from-release-process), will always have the top most entry in
+the `CHANGELOG.md` set to an unreleased version. That top most section will have either
+the minor or the major versions bumped, but not both, when compared to the latest released
+version. Which one it will be depends on the recent changes that were added after the
+latest release was made. More on why this should always be true is in the [Release
+Process](#release-process) section. For example, if the latest version of
+`cardano-ledger-core` that was released to CHaPs is `1.20.1.0`, then there will be a
+section like this:
+
+```
+# Version history for `cardano-ledger-core`
+
+## 1.20.2.0
+
+* Add ...
+...
+```
+
+It is quite common to experience conflicts in the changelog, since that will be the most
+common section of the codebase being updated at the same time. With respect on conflicts
+resolution on the package version it is pretty easy, the highest value should usually win.
+
+With respect to the actual change log entries, there is no enforced order in which entries
+should be added. To the top, to the bottom or in the middle of the list, it is
+irrelevant. In fact, randomized insertion reduces conflicts, so feel free to add entries
+anywhere in the section. Most important is that we can always infer the actual order from
+the git history, which provide with a very reliable source of order on changes.
+
+#### When to update
+
+`CHANGELOG.md` must be updated for the package that is expereincing a change and any other
+package in the repository that re-exports that change in one form or another. In general
+an entry is added according to these rules:
+
+* Every breaking change MUST appear in the `CHANGELOG.md`.
+
+* Every non-breaking change SHOULD appear in the `CHANGELOG.md`
+
+* Every change that does not change the user facing semantics of the code (eg. changes to
+  documentation, depenency bounds, test suites, addition of internal functions,
+  performance improvements, etc.)  COULD be added to the `CHANGELOG.md`, but they are
+  discouraged. Exceptions should be made for changes that could be valueable to the
+  downstream user.
+
+* Every change to a public sub-library of a package must be added to a separate section in
+  the `CHANGELOG.md` and is versioned together with the main library. Eg. `testlib`:
+
+  ```
+  # Version history for `cardano-ledger-core`
+
+  ## 1.20.2.0
+
+  ### `testlib`
+
+  * Add `Arbitrary` instance for ...
+  ...
+  ```
+
+## Release Process
+
+Usually a package release will happen directly from `master` branch. Current release
```

Comment:
We should remove mention of the "trunk based developement" then: https://github.com/input-output-hk/cardano-ledger/blob/master/CONTRIBUTING.md#branching-model

Reply:
I've removed the link and other no longer relevant information. I have left a mention about the trunk based development, since technically that is still our model. We just do releasing differently, since we have many packages that we can release, not just one.

Reply:
ok, thanks!